lib/azuremetadata/azuremetadata.py

```diff
@@ -65,19 +65,46 @@ def get_disk_tag(self, device=None):
             return ''
 
     @staticmethod
-    def _find_root_device():
+    def _find_block_device(mountpoint="/"):
         """Returns detected root device path or None if detection failed."""
+
+        out, err = AzureMetadata._get_lsblk_output()
+
+        if err or not out:
+            return None
+        else:
+            try:
+                data = json.loads(out.decode("utf-8"))
+            except json.decoder.JSONDecodeError:
+                return None
```

Comment:
user doesn't know if the block device is empty because of their config or because of a JSON error, 
maybe the method should log something.